src/github.com/kelseyhightower/confd/backends/etcd/client.go

```diff
@@ -28,10 +30,23 @@ func NewEtcdClient(machines []string, cert, key string, caCert string) (*Client,
 	}
 	// Configure the DialTimeout, since 1 second is often too short
 	c.SetDialTimeout(time.Duration(3) * time.Second)
-	success := c.SetCluster(machines)
-	if !success {
-		return &Client{c}, errors.New("cannot connect to etcd cluster: " + strings.Join(machines, ","))
+
+	maxConnectAttempts := 10
```

Comment:
1 is really low if you have to wait for an etc service to come up.
This should only happen on launch of confd.
Maybe we should crash the process if it doesn't connect because anyway confd won't do anything if it doesn't connect the first time ?